scripts/bulk-access-control.md

```diff
@@ -0,0 +1,101 @@
+
+If you need to align the access conditions of all the items and/or bitstreams in a community or collection, or other selection to new rules defined by the Institution such as making open access all the thesis or restrict access to all the material related to a specific project, etc.
+
+
+## How it Works?
+
+The Bulk Access Control Management script reads a json file containing the details about which resources (items and bitstreams) need to be processed and how to manipulate the access conditions.
+The script then processes the items and bitstreams identified according to the defined mode and specified accessConditions.
+
+## Parameters
+
+The script takes the following parameters:
+
+| Parameter                        | Description                                                                                                                                                                            |
+|----------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `-u`, <br>`--uuid`                | This parameter, mandatory and repeatable, specifies the uuid of the object to process according to processing instruction, specified in the json file (see below).                                                                                                              |
+
+| `-f`, <br>`--file`                | This parameter specifies the json file containing the processing instruction.                                                                                                              |
+| `-h`, <br>`--help`               | This parameter displays help information for the script.                                                                                                                               |
+
+The json file specified with the file parameter has the following structure
+ structured as follow
+```json
+{
+   item: {
+      mode: "replace",
+      accessConditions: [
+          {
+            "name": "openaccess"
+          }
+      ]
+   },
+   bitstream: {
+      constraints: {
+          uuid: [bit-uuid1, bit-uuid2, ..., bit-uuidN],
+      },
+      mode: "add",
+      accessConditions: [
+        {
+         "name": "embargo",
+         "startDate": "2024-06-24T23:59:59.999+0000"
+        },
+        {
+         "name": "network"         
+        }
+      ]
+   }
+}
+```
+* target: is a list of uuid of communities, collections and individual items to process
+* item: the item node contains the processing detail at the item level for the identified resources
+  * mode: can be replace or add. Replace will mean that after processing the items will have only the access conditions that are specified in the accessContitions property. Add will mean that after processing the items will have the specified accessCondition in addition to the ones already defined.
```

Comment:
Shouldn't it also support remove, so you can remove a specific permission

I'm mainly thinking about the use case where someone did use the add, and needs to revert it

Reply:
Hi @benbosman 
this is an interesting idea that can be considered as a future improvement but it is not included in our original requirements https://wiki.lyrasis.org/display/DSPACE/Bulk+access+control+management+-+4Science+proposal . It is not trivial to implement as the bulk access control manage access condition not individual resource policies so we should analyze the best way to support that. Another complexity is related to the fact that removing an access condition you could eventually end with no access condition defined at all, this could be managed inheriting the owning collection policies.